src/dbt_score/rules/generic.py

```diff
@@ -47,7 +47,49 @@ def sql_has_reasonable_number_of_lines(
 def has_example_sql(model: Model) -> RuleViolation | None:
     """The documentation of a model should have an example query."""
     if model.language == "sql":
-        if "```sql" not in model.description:
+        if "```sql" not in (model.description or ""):
             return RuleViolation(
                 "The model description does not include an example SQL query."
             )
+
+@rule
+def has_uniqueness_test(model: Model) -> RuleViolation | None:
+    """Model has uniqueness test for primary key."""
+    if model.config.get("materialized") not in {"table", "incremental"}:
+        return None
+
+    # Extract PK
+    pk_columns = None
```

Comment:
Can it be an empty list, and we only verify pk_columns if the length is >0?
I think this way we can ensure `pk_columns` is always a list